src/windows.rs

```diff
@@ -0,0 +1,464 @@
+};
+
+#[allow(non_snake_case)]
+#[derive(Debug)]
+pub struct Brightness {
+    hPhysicalMonitor: HANDLE,
+    file_handle: HANDLE,
+    device_name: String,
+    device_description: String,
+    device_id: String,
+    device_key: String,
+    device_string: String,
+}
+
+impl Drop for Brightness {
+    fn drop(&mut self) {
+        unsafe {
+            DestroyPhysicalMonitor(self.hPhysicalMonitor);
+            CloseHandle(self.file_handle);
+        }
+    }
+}
+
+#[async_trait]
+impl crate::Brightness for Brightness {
+    async fn device_name(&self) -> Result<String, Error> {
+        Ok(self.device_name.clone())
+    }
+
+    async fn device_info(&self) -> Result<HashMap<String, String>, Error> {
+        Ok(hashmap! {
+            "device_description".to_string() => self.device_description.clone(),
+            "device_id".to_string() => self.device_id.clone(),
+            "device_key".to_string() => self.device_key.clone(),
+            "device_string".to_string() => self.device_string.clone(),
+        })
+    }
+
+    async fn get(&self) -> Result<u32, Error> {
+        let ioctl_query = ioctl_query_supported_brightness(self);
+        return match ioctl_query {
+            Ok(_) => ioctl_query_display_brightness(self).map_err(|e| e.into()),
+            Err(e) if e == ERROR_NOT_FOUND || e == WIN32_ERROR(6) => {
+                ddcci_get_monitor_brightness(self)
+                    .map(|b| b.get_current_percentage())
+                    .map_err(|e| e.into())
+            }
+            Err(e) => Err(SysError::GettingBrightnessFailed {
+                device_name: self.device_name.clone(),
+                function_name: "IOCTL_VIDEO_QUERY_SUPPORTED_BRIGHTNESS",
+                last_error: e,
+            }
+            .into()),
+        };
+    }
+
+    async fn set(&mut self, percentage: u32) -> Result<(), Error> {
+        let ioctl_query = ioctl_query_supported_brightness(self);
+        match ioctl_query {
+            Ok(levels) => {
+                let new_value = levels.get_nearest(percentage);
+                ioctl_set_display_brightness(self, new_value).map_err(|e| e.into())
+            }
+            Err(e) if e == ERROR_NOT_FOUND => {
+                let new_value = ddcci_get_monitor_brightness(self)
+                    .map(|b| b.percentage_to_current(percentage))?;
+                ddcci_set_monitor_brightness(self, new_value).map_err(|e| e.into())
+            }
+            Err(e) => Err(SysError::SettingBrightnessFailed {
+                device_name: self.device_name.clone(),
+                function_name: "IOCTL_VIDEO_QUERY_SUPPORTED_BRIGHTNESS",
+                last_error: e,
+            }
+            .into()),
+        }
+    }
+}
+
+pub fn brightness_devices() -> impl Stream<Item = Result<Brightness, SysError>> {
+    unsafe extern "system" fn enum_monitors(
+        arg1: HMONITOR,
+        _arg2: HDC,
+        _arg3: *mut RECT,
+        arg4: LPARAM,
```

Comment:
A meaningful name indicating what this parameter is (e.g. `monitors`) would be more useful.